src/Host/Client/Impl/Host/RHost.cs

```diff
@@ -230,6 +245,56 @@ public sealed partial class RHost : IDisposable, IRExpressionEvaluator {
             await RespondAsync(request, ct, input);
         }
 
+        public Task<SendBlobResult> SendBlobAsync(byte[] data, CancellationToken ct) {
+            return ct.IsCancellationRequested || _runTask == null || _runTask.IsCompleted
+                ? Task.FromCanceled<SendBlobResult>(new CancellationToken(true))
+                : SendBlobAsyncBackground(data, ct);
+        }
+
+        private async Task<SendBlobResult> SendBlobAsyncBackground(byte[] data, CancellationToken ct) {
+            await TaskUtilities.SwitchToBackgroundThread();
+
+            JArray message;
+            var request =  BlobRequest.MakeCreateBlobRequest(this,out message, 1);
+            _blobRequests[request.Id] = request;
+
+            await SendAsync(message, data, ct);
+            return (await request.CompletionSource.Task).ToSendBlobResult();
+        }
+
+        public Task<GetBlobResult> GetBlobAsync(long[] blobIds, CancellationToken ct) {
+            return ct.IsCancellationRequested || _runTask == null || _runTask.IsCompleted
+                ? Task.FromCanceled<GetBlobResult>(new CancellationToken(true))
+                : GetBlobAsyncBackground(blobIds, ct);
+        }
+
+        private async Task<GetBlobResult> GetBlobAsyncBackground(long[] blobIds, CancellationToken ct) {
+            await TaskUtilities.SwitchToBackgroundThread();
+
+            JArray message;
+            var request = BlobRequest.MakeGetBlobsRequest(this, out message, blobIds);
+            _blobRequests[request.Id] = request;
+
+            await SendAsync(message, ct);
+            return (await request.CompletionSource.Task).ToGetBlobResult();
+        }
+
+        public Task DestroyBlobAsync(long[] blobIds, CancellationToken ct) {
+            return ct.IsCancellationRequested || _runTask == null || _runTask.IsCompleted
+                ? Task.FromCanceled(new CancellationToken(true))
+                : DestroyBlobAsyncBackground(blobIds, ct);
+        }
+
+        private async Task DestroyBlobAsyncBackground(long[] blobIds, CancellationToken ct) {
+            await TaskUtilities.SwitchToBackgroundThread();
+
+            JArray message;
+            var request = BlobRequest.MakeDestroyBlobsRequest(this, out message, blobIds);
+            //  we don't expect a response for this so don't add it to the pending blob requests
```

Comment:
In our terminology, "Destroy" is technically a notification, not a request. It doesn't require a response because it can always be assumed to either succeed (if host is alive and connection is live) or not matter (if host is dead or connection is going down; in the latter case, host will shut down anyway). And since we always use transports that guarantee delivery, we don't need to implement our own ACK scheme.